authentik/policies/expression/evaluator.py

```diff
@@ -44,13 +44,14 @@ def expr_func_user_has_authenticator(
     ) -> bool:
         """Check if a user has any authenticator devices, optionally matching *device_type*"""
         user_devices = devices_for_user(user)
-        if device_type:
-            for device in user_devices:
+        for device in user_devices:
```

Comment:
I think an easier way to fix this would be to `return len(list(user_devices)) > 0`, because like this we always check the entire list